nikola/nikola.py

```diff
@@ -2183,6 +2183,11 @@ def generic_page_renderer(self, lang, post, filters, context=None):
         context['title'] = post.title(lang)
         context['description'] = post.description(lang)
         context['permalink'] = post.permalink(lang)
+        if 'crumbs' not in context:
+            if post.permalink(lang).endswith('/'):
```

Comment:
How does this work if `STRIP_INDEXES = False`?

Reply:
If it's `False`', it's a file (which probably ends with `.html`) and files are dealt with in the else clause.

So for ```posts/foo.html``` we'd end up with
```
context['crumbs'] = [['.', 'posts'], ['#', 'foo.html']]
```

I've also tested `foo/bar/' which leads to
```
context['crumbs'] = [['..', 'foo'], ['#', 'bar']]
```

Hmm, I guess `PRETTY_URLS = True` and `STRIP_INDEXES = False` is not handled ideally (although not wrong either). `foo/bar/index.html` would be:
```
context['crumbs'] = [['..', 'foo'], ['.', 'bar'], ['#', 'index.html']]
```

Is this your concern?  Or did I miss some other case?

Reply:
Yeah, that last one is kinda concerning (although this use case is a strange one).  You can do `permalink.endswith(self.config['INDEX_FILE'])` and then do some extra logic.